### ISSUE TITLE
Testing shared app code

### ISSUE THREAD

**Issue by ramijarrar** [Sanjo/meteor-jasmine#227]
Are there currently any recommended patterns for testing app code that is shared between client and server?

**Comment by Sanjo** [comment]
Not really. If there is no difference in the implementation for client and server, just write the tests either for the server or the client.

For package testing you can add the test file to 'server' and 'client'.

I will keep this use case in mind for a future release.

__Notes for me:__

Testing code that is not under a server/ or client/ folder is executed on the server and client.

**Comment by ramijarrar** [comment]
I tried modifying the regex in the server/client modes to allow for this functionality but the framework will still only allow them to run on one mode at a time. Switching the order in which the testing modes are registered in the registerFrameworks.js file switches the mode that they're run in, so it seems like this is a limitation in the way Velocity loads file. Can you confirm this?

**Comment by ramijarrar** [comment]
@Sanjo Any updates on this?

**Comment by ramijarrar** [comment]
Also, been thinking a little more about your idea of merging the two frameworks into one and one issue that needs to be considered is the implication on reporting. Unless the frameworks are kept separate, the reporter will group shared specs under the same heading, making it impossible to differentiate between their environments.

**Comment by Sanjo** [comment]
> Any updates on this?

Not yet. I will finish the release this week.

Regarding the reporting you are right. Maybe I will use you suggested pull-request for now until we have improved the reporters.

**Comment by samhatoum** [comment]
would a simple symlink solve this issue?

**Comment by ramijarrar** [comment]
Not really because not all shared code lives in the same directory

**Comment by ramijarrar** [comment]
Also, this is a far less hacky solution :P
